### PR TITLE
#409 Implement timeout support to networking Client

### PR DESCRIPTION
**Pull request by mnylen**
### Changes

- Added timeout option to `Auth0` that is passed to `networking.Client` (defaults to 10 seconds)
- Requests made with `networking.Client` timeout after the specified timeout, and any ongoing requests are cancelled using `AbortController` created for each request.
- Exports `TimeoutError`

### References

Please include relevant links supporting this change such as a:

- https://github.com/auth0/react-native-auth0/issues/409

### Testing

Can be tested in e.g. iOS Simulator using Network Link Condition preferences pane. Setting the packet drop ratio to 100 % will cause the requests to never complete. With this change, `TimeoutError` will be thrown after attempting the request for 10 seconds.

- [x] This change adds unit test coverage
- [ ] This change has been tested on the latest version of the platform/language or why not.

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] All existing and new tests complete without errors
- [x] All active GitHub checks have passed

### CODE REVIEW

**Comment by Widcket** [comment]
Hi @mnylen, thanks for raising this PR. I haven't been able to look at it yet, but might get to it by the end of the week.

File: src/networking/__tests__/index.spec.js [hunk 1]
```diff
@@ -233,8 +234,53 @@ describe('client', () => {
...
       ).toMatchSnapshot();
     });
   });
+
+  describe('timeout', () => {
+    const client = new Client({
+      baseUrl,
+      telemetry: {name: 'react-native-auth0', version: '1.0.0'},
+      token: 'a.bearer.token',
+      timeout: 10,
```
Comment (Widcket) [inline src/networking/__tests__/index.spec.js:247]: Please use a lower value.

**Review by Widcket — COMMENTED** [review]
Looks good, just have a nitpick over the timeout value for the test case. Thank you for the PR, we really appreciate you taking the time for doing this.

**Comment by mnylen** [comment]
@Widcket thanks for the feedback. I changed the test timeout. Sorry it took a while. Switched companies and not in a project that uses this anymore and don't have the setup locally on my computer anymore either, but hopefully others can find this change useful. :)

**Comment by Widcket** [comment]
Thanks for coming back and pushing the change!

**Review by Widcket — APPROVED** [review]

**Pull request merged**